app/src/layouts/appLayout/projectSidebar/projectSidebar.jsx

```diff
@@ -33,81 +29,46 @@ import {
+import { AppSidebar } from 'layouts/common/appSidebar';
+import { ExtensionLoader } from 'components/extensionLoader';
 import { urlOrganizationAndProjectSelector } from 'controllers/pages';
 import FiltersIcon from 'common/img/filters-icon-inline.svg';
+import { SidebarButton } from 'components/buttons/sidebarButton/sidebarButton';
 import DashboardIcon from './img/dashboard-icon-inline.svg';
 import LaunchesIcon from './img/launches-icon-inline.svg';
 import DebugIcon from './img/debug-icon-inline.svg';
-import ProfileIcon from './img/profile-icon-inline.svg';
-import AdministrateIcon from './img/administrate-icon-inline.svg';
 import MembersIcon from './img/members-icon-inline.svg';
 import SettingsIcon from './img/settings-icon-inline.svg';
-import { ProjectSelector } from '../../common/projectSelector';
+import { OrganizationsBlock } from './organizationsBlock';
+import { OrganizationsControlWithPopover } from './organizationsControl';
 
-@connect((state) => ({
-  projectName: projectNameSelector(state),
-  availableProjects: availableProjectsSelector(state),
-  projectRole: activeProjectRoleSelector(state),
-  accountRole: userAccountRoleSelector(state),
-  extensions: uiExtensionSidebarComponentsSelector(state),
-  slugs: urlOrganizationAndProjectSelector(state),
-}))
-@track()
-export class AppSidebar extends Component {
-  static propTypes = {
-    projectRole: PropTypes.string.isRequired,
-    projectName: PropTypes.string.isRequired,
-    accountRole: PropTypes.string.isRequired,
-    tracking: PropTypes.shape({
-      trackEvent: PropTypes.func,
-      getTrackingData: PropTypes.func,
-    }).isRequired,
-    availableProjects: PropTypes.object,
-    extensions: PropTypes.arrayOf(extensionType),
-    onClickNavBtn: PropTypes.func,
-    slugs: PropTypes.shape({
-      organizationSlug: PropTypes.string.isRequired,
-      projectSlug: PropTypes.string.isRequired,
-    }),
-  };
-  static defaultProps = {
-    availableProjects: {},
-    extensions: [],
-    onClickNavBtn: () => {},
-  };
+export const ProjectSidebar = ({ onClickNavBtn }) => {
+  const { trackEvent } = useTracking();
+  const projectRole = useSelector(activeProjectRoleSelector);
+  const accountRole = useSelector(userAccountRoleSelector);
+  const extensions = useSelector(uiExtensionSidebarComponentsSelector);
+  const { organizationSlug, projectSlug } = useSelector(urlOrganizationAndProjectSelector);
+  const [isOpenPopover, setIsOpenPopover] = useState(false);
 
-  onClickButton = (eventInfo) => {
-    this.props.onClickNavBtn();
-    this.props.tracking.trackEvent(eventInfo);
+  const onClickButton = (eventInfo) => {
+    onClickNavBtn();
+    trackEvent(eventInfo);
   };
 
-  createTopSidebarItems = () => {
-    const {
-      projectRole,
-      accountRole,
-      onClickNavBtn,
-      extensions,
-      slugs: { organizationSlug, projectSlug },
-    } = this.props;
-
+  const [topSidebarItems, topSidebarControlItems] = (() => {
```

Comment:
Let's not use unnamed immediately invoked functions.
I suggest to create a named function and call here.

Reply:
this approach was used in the code, removed IIFE